Since slackdiff was updated, channel must in fact be specified as Slack channel_id

Due to deprecation of the old Slack API call, slackdiff was [updated recently](https://github.com/ytti/oxidized/commit/711f3e7d50c5e04cbe3be7db50b4321ea19e5332) to use a different API call for file uploads.

Since this change was made, you can no longer use the display name eg "#my-channel-name" for hooks:slack:channel:, and you get errors like this in the Oxidized log:

`ERROR -- : Hook slack (#<SlackDiff:0x00007fa4f6a5de38>) failed (#<Slack::Web::Api::Errors::InvalidArguments response=#<Faraday::Response:0x00007fa4f6310db0 @on_complete_callbacks=[], @env=#<Faraday::Env @method=:post @request_body="channel_id=%23my-channel-name&files=%5B%7B%22id%22%3A%22XXXXXXXXXXXX%22%2C%22title%22%3A%22XXXX.XXXX.XXXXXXXXX.net++XXXXXXXX%22%7D%5D" @url=#<URI::HTTPS https://slack.com/api/files.completeUploadExternal> @request=#<Faraday::RequestOptions (empty)> @request_headers={"Accept"=>"application/json; charset=utf-8", "User-Agent"=>"Slack Ruby Client/2.4.0", "Authorization"=>"Bearer XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX", "Content-Type"=>"application/x-www-form-urlencoded"} @ssl=#<Faraday::SSLOptions verify=true> @response=#<Faraday::Response:0x00007fa4f6310db0 ...> @response_headers={"date"=>"Tue, 17 Dec 2024 18:43:07 GMT", "server"=>"Apache", "vary"=>"Accept-Encoding", "x-slack-req-id"=>"XXXXXXXXXXXXXXXXXXXXXXXXXXXX", "x-content-type-options"=>"nosniff", "x-xss-protection"=>"0", "pragma"=>"no-cache", "cache-control"=>"private, no-cache, no-store, must-revalidate", "expires"=>"Sat, 26 Jul 1997 05:00:00 GMT", "content-type"=>"application/json; charset=utf-8", "x-accepted-oauth-scopes"=>"files:write:user,post", "x-oauth-scopes"=>"identify,read,post,client,apps", "access-control-expose-headers"=>"x-slack-req-id, retry-after", "access-control-allow-headers"=>"slack-route, x-slack-version-ts, x-b3-traceid, x-b3-spanid, x-b3-parentspanid, x-b3-sampled, x-b3-flags", "strict-transport-security"=>"max-age=31536000; includeSubDomains; preload", "referrer-policy"=>"no-referrer", "x-slack-unique-id"=>"XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX", "x-slack-backend"=>"r", "access-control-allow-origin"=>"*", "via"=>"1.1 slack-prod.tinyspeck.com, envoy-www-iad-efpzdarl,envoy-edge-lhr-syfpybig", "content-encoding"=>"gzip", "content-length"=>"162", "x-envoy-attempt-count"=>"1", "x-envoy-upstream-service-time"=>"90", "x-backend"=>"main_normal main_canary_with_overflow main_control_with_overflow", "x-server"=>"slack-www-hhvm-main-iad-bihl", "x-slack-shared-secret-outcome"=>"no-match", "x-edge-backend"=>"envoy-www", "x-slack-edge-shared-secret-outcome"=>"no-match"} @status=200 @reason_phrase="OK" @response_body=#<Slack::Messages::Message error="invalid_arguments" ok=false response_metadata=#<Slack::Messages::Message messages=#<Hashie::Array ["[ERROR] input must match regex pattern: ^[CGDZ][A-Z0-9]{8,}$ [json-pointer:/channel_id]"]>>>>>>) for event :post_store`

the destination channel must now be specified using the slack "channel_id" which is a type identifier (C/G/D/Z) followed by a random base-36 number generated by slack when the channel is created (matching the regex ^[CGDZ][A-Z0-9]{8,}$) e.g.

```
hooks:
  slack:
    channel: GA1B2C3D4E5
```

The oxidized documentation still says that you should specify "#my-channel-name" and this should be changed to reflect the fact that a channel_id is now expected. 

Perhaps the variable name should also be changed from channel to channel_id to be explicit though this will of course require the commensurate code change to slackdiff.rb as well.

I did wonder if the effort should be made to preserve the old behaviour by making an intermediate API call to try to resolve a channel display name to a channel_id, but it seems that Slack guidance is that channel_id should be explicitly used for this sort of use case because some time after slackdiff is configured the channel could subsequently be renamed by someone and using channel_id ensures preservation of the target channel (and so, by proxy, the slack users who have access to that channel and can see that content) based on the originally configured channel rather than whatever channel currently holds that display name, so it seems like channel_id is the better choice to ensure deterministic behaviour of where the messages go.

--- a/docs/Hooks.md
+++ b/docs/Hooks.md
@@ -259,7 +259,7 @@ gem install slack-ruby-client
 
 ### slackdiff hook configuration example
 
-> Please note that channel needs to be your Slack channel ID.
+> Please note that the channel needs to be your Slack channel ID.
 
 ```yaml
 hooks:
@@ -284,8 +284,6 @@ hooks:
     diff: false
     message: "%{node} %{group} %{model} updated https://git.intranet/network-changes/commit/%{commitref}"
 ```
-
-Note the channel ID must be in quotes.
 
 A proxy can optionally be specified if needed to reach the Slack API endpoint.
 

--- a/docs/Hooks.md
+++ b/docs/Hooks.md
@@ -259,13 +259,15 @@ gem install slack-ruby-client
 
 ### slackdiff hook configuration example
 
+> Please note that channel needs to be your Slack channel ID.
+
 ```yaml
 hooks:
   slack:
     type: slackdiff
     events: [post_store]
     token: SLACK_BOT_TOKEN
-    channel: "#network-changes"
+    channel: "CHANNEL_ID"
 ```
 
 The token parameter is a Slack API token that can be generated following [this tutorial](https://api.slack.com/tutorials/tracks/getting-a-token).  Until Slack stops supporting them, legacy tokens can also be used.
@@ -278,12 +280,12 @@ hooks:
     type: slackdiff
     events: [post_store]
     token: SLACK_BOT_TOKEN
-    channel: "#network-changes"
+    channel: "CHANNEL_ID"
     diff: false
     message: "%{node} %{group} %{model} updated https://git.intranet/network-changes/commit/%{commitref}"
 ```
 
-Note the channel name must be in quotes.
+Note the channel ID must be in quotes.
 
 A proxy can optionally be specified if needed to reach the Slack API endpoint.
 
@@ -293,7 +295,7 @@ hooks:
     type: slackdiff
     events: [post_store]
     token: SLACK_BOT_TOKEN
-    channel: "#network-changes"
+    channel: "#CHANNEL_ID"
     proxy: http://myproxy:8080
 ```
 

